Explorer/Assets/Scripts/ECS/ComponentsPooling/ComponentPool.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using UnityEngine.Pool;
+using Utility.ThreadSafePool;
+
+namespace ECS.ComponentsPooling
+{
+    public class ComponentPool<T> : IComponentPool<T> where T: class, new()
```

Comment:
Should these base systems have test coverage?

Reply:
Should, I will add